File: backend/src/main/java/cn/edu/nynu/codelab/site/entity/SiteConfig.java
package cn.edu.nynu.codelab.site.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 站点配置实体
 *
 * @author NYNU Code Lab
 */
@Data
@TableName("lab_site_config")
public class SiteConfig {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String configKey;

    private String configValue;

    private String configType;

    private String groupName;

    private String remark;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: admin-web/src/api/siteConfig.ts
import request from './request'

export interface SiteConfigItem {
  id: number
  configKey: string
  configValue: string
  configType: string
  groupName: string
  remark: string
  createTime: string
  updateTime: string
}

export interface SiteConfigUpdateParams {
  configKey: string
  configValue: string
  configType?: string
  groupName?: string
  remark?: string
}

export function getSiteConfigs(): Promise<{ code: number; data: SiteConfigItem[]; message?: string }> {
  return request.get('/admin/site-config')
}

export function updateSiteConfig(key: string, data: SiteConfigUpdateParams): Promise<{ code: number; data: SiteConfigItem; message?: string }> {
  return request.put(`/admin/site-config/${encodeURIComponent(key)}`, data)
}
